danascheider/skyrim_inventory_management_frontend
# Fix up and test shopping list destroy functionality

## Context

[**Test shopping list destroy functionality**](https://trello.com/c/aqkppCpX/111-test-shopping-list-destroy-functionality)
[**Add test cases for shopping list deletion**](https://trello.com/c/bTk9Bt2g/84-add-test-cases-for-shopping-list-deletion)

After adding the Game model, we've had to rework the shopping lists page a bit to accommodate the fact that shopping lists now belong to games. The backend route for shopping list deletion was not affected, so the primary task here was to write automated tests for the functionality and then fix up a few things that had broken with recent refactoring. 

## Changes

Summary of the changes made, preferably as a bulleted list

## Considerations

* Add Jest tests for shopping list destroy functionality
* Make a few changes to components and shopping lists context provider to make sure flash messages are displayed and components are properly mounted/unmounted
* Update Storybook stories to allow for delete API calls
* Update docs on shopping lists context to reflect new APIs in functions (callback objects instead of positional args)
* Add docs on common Jest failures/issues

## Manual Test Cases

This doesn't change behaviour from what it was previously - except for the fact that the shopping lists displayed are now scoped to the games that own them. So just make sure that things still work like before when a list is deleted. Also test that lists are not deleted if you cancel when prompted.
